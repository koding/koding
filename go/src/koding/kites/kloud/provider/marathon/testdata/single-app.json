{
  "resource": {
    "marathon_app": {
      "app": {
        "cmd": "python3 -m http.server 8080",
        "container": {
          "docker": [
            {
              "image": "python:3",
              "network": "BRIDGE"
            }
          ]
        },
        "cpus": 1.2,
        "mem": 256
      }
    }
  }
}
