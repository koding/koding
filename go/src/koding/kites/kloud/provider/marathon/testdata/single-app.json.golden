{
  "resource": {
    "marathon_app": {
      "app": {
        "app_id": "/app-foobar-ident",
        "container": [
          {
            "docker": [
              {
                "image": "python:3",
                "network": "BRIDGE",
                "parameters": {
                  "parameter": [
                    {
                      "key": "entrypoint",
                      "value": "/mnt/mesos/sandbox/entrypoint.${count.index * 1 + 1}.sh"
                    }
                  ]
                },
                "port_mappings": {
                  "port_mapping": [
                    {
                      "container_port": 56789,
                      "host_port": 0,
                      "protocol": "tcp"
                    }
                  ]
                }
              }
            ]
          }
        ],
        "count": 1,
        "cpus": 1.2,
        "env": {
          "KODING_CMD": "${base64encode(null_resource.app.triggers.cmd)}",
          "KODING_KLIENT_URL": "$KLIENT_URL",
          "KODING_METADATA_1": "..."
        },
        "fetch": [
          {
            "cache": false,
            "executable": false,
            "uri": "$SCREEN_URL"
          },
          {
            "cache": true,
            "executable": false,
            "uri": "$CERT_URL"
          },
          {
            "cache": false,
            "executable": false,
            "uri": "$KLIENT_URL"
          },
          {
            "cache": false,
            "executable": true,
            "uri": "$ENTRYPOINT_URL/entrypoint.1.sh"
          }
        ],
        "mem": 256,
        "ports": [
          0
        ]
      }
    },
    "null_resource": {
      "app": {
        "depends_on": [],
        "triggers": {
          "cmd": "python3 -m http.server 8080"
        }
      }
    }
  }
}
